sled-agent/src/services.rs

```diff
@@ -927,11 +927,34 @@ impl ServiceManager {
         needed
     }
 
+    /// Enable DNS resolution using the internal DNS servers for the given zone.
+    ///
+    /// The Internal DNS servers are allocated on a reserved AZ-wide subnet which we can
+    /// determine from the given Sled underlay address.
+    ///
+    /// For more details on this "reserved" rack subnet, refer to
+    /// [omicron_common::address::ReservedRackSubnet].
+    async fn configure_internal_dns_in_zone(
+        &self,
+        running_zone: &RunningZone,
+        sled_addr: Ipv6Addr,
+    ) -> Result<(), Error> {
+        let az_services_subnet =
+            ReservedRackSubnet::new(Ipv6Subnet::<AZ_PREFIX>::new(sled_addr));
+        let dns_servers = az_services_subnet
+            .get_dns_subnets()
+            .into_iter()
+            .map(|dns_subnet| dns_subnet.dns_address().ip().into())
+            .collect::<Vec<_>>();
+
+        self.configure_dns_client(running_zone, &dns_servers, None).await
```

Comment:
I think it's great to set up /etc/resolv.conf in most zones.  But I think we don't want to use `configure_dns_client()` to do it.  I summarized here:
https://github.com/oxidecomputer/omicron/issues/2122#issuecomment-1611870486

I think what we did for the CockroachDB zone is a more robust approach:
https://github.com/oxidecomputer/omicron/blob/b514ef8b8659fb043b804c88a827bf321e3cb58b/sled-agent/src/services.rs#L1100-L1130

I am not sure if that's going to require the other bits mentioned in https://github.com/oxidecomputer/omicron/issues/2122#issuecomment-1611889979 though.